Source/Common/Profiler.h

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+/*
+$info$
+tags: Common|Profiler
+desc: Frontend profiler common code
+$end_info$
+*/
+#pragma once
+#include <FEXCore/Utils/Profiler.h>
+
+namespace FEXCore::Core {
+struct InternalThreadState;
+}
+
+#ifdef _M_ARM_64
+static inline void memory_barrier() {
+  asm volatile("dmb ishst;" ::: "memory");
```

Comment:
can atomic_thread_fence not be used here? instead of inline asm

Reply:
That helper can't actually generate a `dmb ishst`, it can only generate `dmb ish` or `dmb ishld`. Sadly.

Reply:
ah, this should ideally be called something more explicit then

Reply:
renamed to `store_memory_barrier`